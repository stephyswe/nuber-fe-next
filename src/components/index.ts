export { UnStyledButton as Button } from './buttons/UnStyledButton';
export { NextImage } from './image';
export { ButtonInput, Input } from './inputs';
export { Link } from './links';
export { default as Typography } from './typography';
