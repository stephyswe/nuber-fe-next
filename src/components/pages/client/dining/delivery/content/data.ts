export const deliveryContentData = [
  {
    type: 'item',
    title: 'In a rush?',
    subtitle: 'Hereâ€™s the fastest delivery for you',
  },
  {
    type: 'item',
    title: 'Only on Uber Eats',
  },
  {
    type: 'section',
    title: 'Pick it up for free',
    subtitle: 'Skip the fees when you order pickup',
    btnText: 'See map',
    image:
      'https://d4p17acsd5wyj.cloudfront.net/eatsfeed/pickup-homefeed-carousel/pickupcarousel_desktopweb.svg',
  },
  {
    type: 'item',
    title: 'Popular near you',
  },
  {
    type: 'item',
    title: 'New on Uber Eats',
    subtitle: 'Be one of the first to support them',
  },
  {
    type: 'section',
    title: 'Free nationwide shipping',
    subtitle: 'Gourmet eats, shipped free',
    image:
      'https://d4p17acsd5wyj.cloudfront.net/eatsfeed/shipment/shipment_carousel_background_image_desktop_web.svg',
  },
];
