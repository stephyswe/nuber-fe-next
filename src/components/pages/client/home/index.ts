export { CountryList, CountryWithMap } from './country';
export { EmploymentList } from './employment';
