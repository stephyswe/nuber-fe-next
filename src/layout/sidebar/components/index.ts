export { SidebarAuthentication } from './authentication';
export { SidebarLinkList } from './link';
export { SidebarSocialList } from './social';
