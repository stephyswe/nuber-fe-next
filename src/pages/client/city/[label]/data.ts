const cityInfo = [
  {
    title: 'Göteborg food delivery and takeout',
    subtitle:
      'With 80 restaurants in Göteborg on Uber Eats, including 12Till12Gott, Stigbergs Pizzeria, and DINÉ Burgers - Femman , you’ll have your pick of places from which to order food online. Get food, from Fast Food to Breakfast And Brunch, from some of the best restaurants in Göteborg delivered to your door. If you’d prefer to get your takeout yourself, simply browse Göteborg restaurants offering pickup.',
  },
  {
    title: 'Göteborg restaurants that deliver',
    subtitle:
      'Uber Eats helps you find food delivery and pickup options from a wide selection of places to eat in Göteborg. Enter an address to browse Göteborg restaurants and cafes offering food delivery. See Göteborg restaurants on Uber Eats that you’ve never tried? View their menus and star ratings to help decide if you’d like to try their food.',
  },
  {
    title: 'Best food in Göteborg',
    subtitle:
      'On a quest to taste the best food in Göteborg? Search for famous restaurants in Göteborg or for your personal favorite places to eat in Göteborg to see if they offer food delivery with Uber Eats. Sometimes the best food is just what you’re craving so if you know what you’d like to eat, browse Göteborg restaurants that deliver by cuisine or dish.',
  },
];

const breadcrumb = [
  {
    title: 'Sweden',
    link: '/',
  },
  {
    title: 'Västra Götaland',
    link: '/client/region/västra-götaland',
  },
  {
    title: 'Göteborg',
    link: '/client/city/göteborg-västra-götaland',
  },
];

export const mockDataCity = {
  title: 'Food Delivery in Göteborg',
  breadcrumb,
  cityInfo,
};
