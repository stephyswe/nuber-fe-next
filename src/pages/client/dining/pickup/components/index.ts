export {
  CategoryContentSkeleton,
  PickupArrow,
  PickupCategoryList,
} from './category';
export { PickupFilter } from './filter';
export { PickupRestaurantList } from './store';
