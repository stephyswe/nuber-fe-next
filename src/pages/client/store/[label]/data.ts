export const dishNewData = [
  {
    __typename: 'Dish',
    id: 1,
    name: 'Pizza Speciale',
    price: 115,
    photo: '',
    description: 'Juicy!',
    type: 'Pizza',
    options: [],
  },
  {
    __typename: 'Dish',
    id: 2,
    name: 'Räksallad',
    price: 115,
    photo: '',
    description:
      'Räkor, vitost, ägg, citron, grekiska kalamataoliver. Isbergssallad, färska tomater, gurka, lök, vitost och bröd ingår.',
    type: 'Sallader',
    options: [],
  },
  {
    __typename: 'Dish',
    id: 3,
    name: 'KycklingSallad',
    price: 115,
    photo:
      'https://tb-static.uber.com/prod/image-proc/processed_images/fbf500543dbcc1a91bcb8de08a66396f/ffd640b0f9bc72838f2ebbee501a5d4b.jpeg',
    description:
      'Kyckling, vitost, färsk paprika, ananas. Isbergssallad, färska tomater, gurka, lök, vitost och bröd ingår.',
    type: 'Sallader',
    options: [
      {
        __typename: 'DishOption',
        name: 'Spice Level',
        extra: null,
        choices: [
          {
            __typename: 'DishChoice',
            name: 'Little Bit',
            extra: null,
          },
          {
            __typename: 'DishChoice',
            name: 'Strong',
            extra: null,
          },
        ],
      },
      {
        __typename: 'DishOption',
        name: 'Pickle',
        extra: 1,
        choices: null,
      },
      {
        __typename: 'DishOption',
        name: 'Size',
        extra: null,
        choices: [
          {
            __typename: 'DishChoice',
            name: 'L',
            extra: 2,
          },
          {
            __typename: 'DishChoice',
            name: 'XL',
            extra: 5,
          },
        ],
      },
    ],
  },
];

export const dishData = {
  Pizza: [
    {
      __typename: 'Dish',
      id: 1,
      name: 'Pizza Speciale',
      price: 115,
      photo: '',
      description: 'Juicy!',
      type: 'Pizza',
      options: [],
    },
  ],
  Sallader: [
    {
      __typename: 'Dish',
      id: 2,
      name: 'Räksallad',
      price: 115,
      photo: '',
      description:
        'Räkor, vitost, ägg, citron, grekiska kalamataoliver. Isbergssallad, färska tomater, gurka, lök, vitost och bröd ingår.',
      type: 'Sallader',
      options: [],
    },
    {
      __typename: 'Dish',
      id: 3,
      name: 'KycklingSallad',
      price: 115,
      photo:
        'https://tb-static.uber.com/prod/image-proc/processed_images/fbf500543dbcc1a91bcb8de08a66396f/ffd640b0f9bc72838f2ebbee501a5d4b.jpeg',
      description:
        'Kyckling, vitost, färsk paprika, ananas. Isbergssallad, färska tomater, gurka, lök, vitost och bröd ingår.',
      type: 'Sallader',
      options: [
        {
          __typename: 'DishOption',
          name: 'Spice Level',
          extra: null,
          choices: [
            {
              __typename: 'DishChoice',
              name: 'Little Bit',
              extra: null,
            },
            {
              __typename: 'DishChoice',
              name: 'Strong',
              extra: null,
            },
          ],
        },
        {
          __typename: 'DishOption',
          name: 'Pickle',
          extra: 1,
          choices: null,
        },
        {
          __typename: 'DishOption',
          name: 'Size',
          extra: null,
          choices: [
            {
              __typename: 'DishChoice',
              name: 'L',
              extra: 2,
            },
            {
              __typename: 'DishChoice',
              name: 'XL',
              extra: 5,
            },
          ],
        },
      ],
    },
  ],
};
