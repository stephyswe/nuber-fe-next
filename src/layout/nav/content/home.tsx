import clsx from 'clsx';

import { useWindowSizeJs } from '@/hooks/useWindowSizeJs';

import { ButtonInput } from '@/components';
import { LinkLogin, LinkSignup } from '@/components/links/NavLink';

import { navHomeData } from '@/constant/pages/layout/nav-home.data';
import { Spacer } from '@/ui';
import { SvgMap } from '@/ui/icons';

export const NavHomeContent = ({ onScroll, change }: any) => {
  const { isMobile } = useWindowSizeJs();
  const { auth } = navHomeData;

  if (isMobile) {
    return (
      <>
        {!change ? (
          <>
            <div className={clsx('relative w-full max-w-[722px]')}></div>
            <div className='flex-1'></div>
            <AuthContent data={auth} change={change} small />
          </>
        ) : (
          <div className='w-full'>
            <ButtonInput
              svg={<SvgMap />}
              placeholder='Enter delivery address'
              className='bg-[#eee]'
              innerClassName='bg-[#eee]'
            />
          </div>
        )}
      </>
    );
  }

  return (
    <>
      <Spacer className='w-[128px]' />
      <div className={clsx('relative w-full max-w-[722px]')}>
        <NavButtonInput onScroll={onScroll} />
      </div>
      <div className='flex-1'></div>
      <Spacer className='w-[24px]' />
      <AuthContent data={auth} change={change} />
    </>
  );
};

export const NavButtonInput = ({ onScroll }: any) => (
  <div
    style={{
      transformOrigin: 'left',
      transition: 'transform 500ms',
      transform: `${onScroll ? 'scaleX(1)' : 'scaleX(0)'}`,
    }}
    className='flex-1'
  >
    {onScroll ? (
      <ButtonInput
        svg={<SvgMap />}
        placeholder='Enter delivery address'
        className='bg-[#eee]'
        innerClassName='bg-[#eee]'
      />
    ) : null}
  </div>
);

type AuthContentProps = {
  data: {
    register: { title: string; link: string };
    login: { title: string; svg: JSX.Element; link: string };
  };
  change: boolean;
  small?: boolean;
};

const AuthContent = ({
  data: { login, register },
  change,
  small,
}: AuthContentProps) => (
  <>
    <LinkLogin
      textHidden
      title={login.title}
      href='/auth/login'
      size={small ? 'small' : 'normal'}
      className={clsx(
        'bg-white text-black hover:bg-[#ccc]',
        change ? 'box-shadow-rgb-double ' : ''
      )}
    />
    <Spacer className='w-4' />
    <LinkSignup
      href='/auth/login'
      title={register.title}
      size={small ? 'small' : 'normal'}
      className={clsx(
        'bg-black text-white hover:bg-[#333]',
        change ? 'box-shadow-rgb-double ' : ''
      )}
    />
  </>
);
