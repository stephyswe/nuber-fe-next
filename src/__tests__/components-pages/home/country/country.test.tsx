import { render, screen, takeSnapshot } from 'test-utils';

import { CountryWithMap } from '@/components/pages';

import { HomeData } from '@/constant/pages/home.data';

const defaultData = {
  title: '',
  link: {
    href: '/',
    title: '',
  },
  mapCenter: {
    lat: 0,
    lng: 0,
  },
  cities: [],
};

// ** Snapshot ** //

takeSnapshot(<CountryWithMap data={defaultData} />);
takeSnapshot(<CountryWithMap data={HomeData.countryData} />);

// ** Default ** //

test('container has elements', () => {
  render(<CountryWithMap data={defaultData} />);
  const element = screen.getByTestId('ui-country-with-map');
  expect(element).toBeInTheDocument();

  // one text element
  const typography = screen.getByTestId('ui-country-with-map-typography');
  expect(typography).toHaveTextContent(defaultData.title);

  // one link element
  const link = screen.getByTestId('ui-country-with-map-link');
  expect(link).toHaveTextContent(defaultData.link.title);
  expect(link).toHaveAttribute('href', defaultData.link.href);

  // one map element
  expect(screen.getByTestId('ui-map-home')).toBeInTheDocument();

  // one country list element
  expect(screen.getByTestId('ui-country-with-map-list')).toBeInTheDocument();
});

// ** Props - Item - Data ** //

describe('props', () => {
  test('data', () => {
    const { link, title } = HomeData.countryData;

    render(<CountryWithMap data={HomeData.countryData} />);

    // text and link in list
    const typography = screen.getByTestId('ui-country-with-map-typography');
    expect(typography).toHaveTextContent(title);

    // one link element
    const linkEl = screen.getByTestId('ui-country-with-map-link');
    expect(linkEl).toHaveTextContent(link.title);
    expect(linkEl).toHaveAttribute('href', link.href);

    // one map element
    expect(screen.getByTestId('ui-map-home')).toBeInTheDocument();

    // one country list element
    expect(screen.getByTestId('ui-country-with-map-list')).toBeInTheDocument();
  });
});

//TODO: add more tests for this component
