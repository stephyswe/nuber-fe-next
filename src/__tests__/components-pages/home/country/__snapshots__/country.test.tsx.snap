// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-testid="ui-country-with-map"
>
  <div
    className="flex items-end justify-between"
  >
    <h2
      className="font-uberMove text-2xl md:text-4xl md:leading-[44px] font-bold"
      data-testid="ui-country-with-map-typography"
    />
    <a
      className="text-base leading-5 font-medium invisible underline md:visible"
      data-testid="ui-country-with-map-link"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    />
  </div>
  <div
    className="pt-6"
  >
    <div
      className="h-[360px] w-full"
      data-testid="ui-map-home"
      id="homeMap"
    >
      <div
        onClick={[Function]}
        onMouseDownCapture={[Function]}
        onMouseMove={[Function]}
        style={
          Object {
            "height": "100%",
            "margin": 0,
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "left": 0,
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        />
        <div
          style={
            Object {
              "height": "50%",
              "left": "50%",
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "top": "50%",
              "width": "50%",
            }
          }
        >
          <div
            style={
              Object {
                "height": "100%",
                "left": 0,
                "margin": 0,
                "padding": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="mt-6 mb-4 grid grid-cols-2 gap-6 md:mb-0 md:grid-cols-4"
    data-testid="ui-country-with-map-list"
  />
</div>
`;

exports[`renders correctly 2`] = `
<div
  data-testid="ui-country-with-map"
>
  <div
    className="flex items-end justify-between"
  >
    <h2
      className="font-uberMove text-2xl md:text-4xl md:leading-[44px] font-bold"
      data-testid="ui-country-with-map-typography"
    >
      Cities near me
    </h2>
    <a
      className="text-base leading-5 font-medium invisible underline md:visible"
      data-testid="ui-country-with-map-link"
      href="/se/location"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      View all +500 cities
    </a>
  </div>
  <div
    className="pt-6"
  >
    <div
      className="h-[360px] w-full"
      data-testid="ui-map-home"
      id="homeMap"
    >
      <div
        onClick={[Function]}
        onMouseDownCapture={[Function]}
        onMouseMove={[Function]}
        style={
          Object {
            "height": "100%",
            "margin": 0,
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "left": 0,
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        />
        <div
          style={
            Object {
              "height": "50%",
              "left": "50%",
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "top": "50%",
              "width": "50%",
            }
          }
        >
          <div
            style={
              Object {
                "height": "100%",
                "left": 0,
                "margin": 0,
                "padding": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <div
              style={
                Object {
                  "backgroundColor": "transparent",
                  "height": 0,
                  "left": -151.6129848888886,
                  "position": "absolute",
                  "top": -32.54057380991617,
                  "width": 0,
                }
              }
            >
              <a
                className="box-border block h-[30px] w-[30px] rounded-[50%]
       transition-all-300 transform-3d box-shadow-map-12 hover:transform-hover-map-home css-9dbljh"
                href="/client/city/g%C3%B6teborg-v%C3%A4stra-g%C3%B6taland"
              >
                <div
                  className="flex h-full w-full items-center justify-center rounded-[50%] bg-white"
                >
                  <div
                    className="h-4 w-4 leading-[1]"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="16px"
                      viewBox="0 0 24 24"
                      width="16px"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.5834 5.16602C14.5001 2.08268 9.50008 2.08268 6.41675 5.16602C3.33341 8.24935 3.33341 13.3327 6.41675 16.416L12.0001 21.9993L17.5834 16.3327C20.6667 13.3327 20.6667 8.24935 17.5834 5.16602ZM12.0001 12.416C11.0834 12.416 10.3334 11.666 10.3334 10.7493C10.3334 9.83268 11.0834 9.08268 12.0001 9.08268C12.9167 9.08268 13.6667 9.83268 13.6667 10.7493C13.6667 11.666 12.9167 12.416 12.0001 12.416Z"
                        fill="#000000"
                      />
                    </svg>
                  </div>
                </div>
              </a>
            </div>
            <div
              style={
                Object {
                  "backgroundColor": "transparent",
                  "height": 0,
                  "left": -112.01831822222084,
                  "position": "absolute",
                  "top": 141.52276928542688,
                  "width": 0,
                }
              }
            >
              <a
                className="box-border block h-[30px] w-[30px] rounded-[50%]
       transition-all-300 transform-3d box-shadow-map-12 hover:transform-hover-map-home css-4403v7"
                href="/client/city/malmö-skåne"
              >
                <div
                  className="flex h-full w-full items-center justify-center rounded-[50%] bg-white"
                >
                  <div
                    className="h-4 w-4 leading-[1]"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="16px"
                      viewBox="0 0 24 24"
                      width="16px"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.5834 5.16602C14.5001 2.08268 9.50008 2.08268 6.41675 5.16602C3.33341 8.24935 3.33341 13.3327 6.41675 16.416L12.0001 21.9993L17.5834 16.3327C20.6667 13.3327 20.6667 8.24935 17.5834 5.16602ZM12.0001 12.416C11.0834 12.416 10.3334 11.666 10.3334 10.7493C10.3334 9.83268 11.0834 9.08268 12.0001 9.08268C12.9167 9.08268 13.6667 9.83268 13.6667 10.7493C13.6667 11.666 12.9167 12.416 12.0001 12.416Z"
                        fill="#000000"
                      />
                    </svg>
                  </div>
                </div>
              </a>
            </div>
            <div
              style={
                Object {
                  "backgroundColor": "transparent",
                  "height": 0,
                  "left": 124.63945955555755,
                  "position": "absolute",
                  "top": -173.66638353387134,
                  "width": 0,
                }
              }
            >
              <a
                className="box-border block h-[30px] w-[30px] rounded-[50%]
       transition-all-300 transform-3d box-shadow-map-12 hover:transform-hover-map-home css-1f0hkc0"
                href="/client/city/stockholm-stockholm"
              >
                <div
                  className="flex h-full w-full items-center justify-center rounded-[50%] bg-white"
                >
                  <div
                    className="h-4 w-4 leading-[1]"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="16px"
                      viewBox="0 0 24 24"
                      width="16px"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.5834 5.16602C14.5001 2.08268 9.50008 2.08268 6.41675 5.16602C3.33341 8.24935 3.33341 13.3327 6.41675 16.416L12.0001 21.9993L17.5834 16.3327C20.6667 13.3327 20.6667 8.24935 17.5834 5.16602ZM12.0001 12.416C11.0834 12.416 10.3334 11.666 10.3334 10.7493C10.3334 9.83268 11.0834 9.08268 12.0001 9.08268C12.9167 9.08268 13.6667 9.83268 13.6667 10.7493C13.6667 11.666 12.9167 12.416 12.0001 12.416Z"
                        fill="#000000"
                      />
                    </svg>
                  </div>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mt-6 mb-4 grid grid-cols-2 gap-6 md:mb-0 md:grid-cols-4"
    data-testid="ui-country-with-map-list"
  >
    <a
      className=""
      data-testid="component-link"
      href="/client/city/g%C3%B6teborg-v%C3%A4stra-g%C3%B6taland"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Gothenburg
    </a>
    <a
      className=""
      data-testid="component-link"
      href="/client/city/malm%C3%B6-sk%C3%A5ne"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Malmö
    </a>
    <a
      className=""
      data-testid="component-link"
      href="/client/city/stockholm-stockholm"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Stockholm
    </a>
  </div>
</div>
`;
